tools/src/wyvern/tools/typedAST/core/declarations/DefDeclaration.java

```diff
@@ -65,8 +73,19 @@ public DefDeclaration(String name, Type returnType, List<NameBinding> argNames,
 		this.argNames = argNames;
 		this.isClass = isClassDef;
 		this.location = location;
+        this.genTypeName = genType;
 	}
 
+	public DefDeclaration(String name, Type returnType, List<NameBinding> argNames,
```

Comment:
Can you make the old constructor call the new one (with a default value for the additional parameter)?